Classes/EventStore/AbstractEventSourcedRepository.php

```diff
@@ -88,15 +85,9 @@ public function __construct()
     final public function save(EventSourcedAggregateRootInterface $aggregate, int $expectedVersion = null)
     {
         $streamName = $this->streamNameResolver->getStreamNameForAggregate($aggregate);
-        $uncommittedEvents = $aggregate->pullUncommittedEvents();
-        $version = $aggregate->getReconstitutionVersion() + 1;
-        $uncommittedEventsWithMetadata = array_map(function (EventInterface $event) use (&$version) {
-            return new EventWithMetadata($event, new EventMetadata([EventMetadata::VERSION => $version ++]));
```

Comment:
> we basically removed the whole concept of aggregate version

Not at all:
So the aggregate is reconstituted from an event stream and knows what version that stream corresponds to (`reconstitutionVersion`).
During it's lifetime it stores it's event in a deterministic order.
When it's saved back to the repository we make sure that no other event has been published to the same aggregate stream in the meantime by comparing the version and if so, we store the new events – in their right order – in one transaction.
Where do you see a missing concept here?

Previously we assumed that if the aggregate is on version 5 and has recorded 3 events that the resulting version is 8. But that must not be true!